tasks/main_services.yml

```diff
@@ -3,6 +3,52 @@
   become: true
   import_tasks: data_volumes.yml
 
+- name: create applicationserver_env dictionary
+  set_fact:
+    base_server_env:
+      VIRTUAL_PORT: "8000"
+      DJANGO_ALLOWED_HOSTS: "{{ hub_configuration.hostname | mandatory }}"
+      DJANGO_SECRET_KEY: "{{ hub_configuration.secret_key | mandatory }}"
+      FROM_EMAIL: "{{ hub_configuration.admin_email | mandatory }}"
+      LETSENCRYPT_HOST: "{{ hub_configuration.hostname | mandatory }}"
+      RAVEN_DSN: "{{ hub_configuration.sentry_dsn | default('') }}"
+      GOOGLE_ANALYTICS_TRACKING_ID: "{{ hub_configuration.google_analytics_id | default('') }}"
+      HUB_OFFERING: "{{ hub_configuration.hub_offering | mandatory }}"
+      OAUTH2_ALLOW_CONCURRENT_ACCESS_TOKENS: "{{ hub_configuration.concurrent_access_tokens | bool | mandatory | string }}"
+
+- name: Upload JWT Server certificates
+  copy:
+    src: "{{ jwt_local_key_path }}/"
+    dest: "{{ jwt_key_path }}/"
+  when:
+  - jwt_key_path | default('', true) | trim | length > 0
+  - jwt_local_key_path | default('', true) | trim | length > 0
+
+- name: Add jwt keys
+  set_fact:
+    base_server_env: "{{ base_server_env | combine ({
+      'JWT_PRIVATE_KEY_PATH': jwt_key_path + '/' + jwt_privatekey,
+      'JWT_PUBLIC_KEY_PATH': jwt_key_path + '/' + jwt_publickey
+      }) }}"
+  when:
+  - jwt_key_path | default('', true) | trim | length > 0
+  - jwt_local_key_path | default('', true) | trim | length > 0
+
+- name: Create application server envs
```

Comment:
1. no abbreviations, clear log messages of what's happening are key to understand what's wrong.
1. more elegant solution with less json fumbling

```yaml
- name: Prepare environment variables for main application server
  set_fact:
    application_server_env: "{{ base_server_env | combine (application_server_specific_envs) }}"
  vars:
    application_server_specific_envs:
      VIRTUAL_PROTO: uwsgi
      VIRTUAL_HOST: "{{ hub_configuration.hostname | mandatory }}"
```

Reply:
any step can have step specific vars. This also means, we can ditch `set_fact` and instead do all of the mambojamo directly on the `docker_container` step like: 

```yaml
- name: Start Django Application Server
  vars: 
    application_server_specific_env:
      VIRTUAL_PROTO: uwsgi
      VIRTUAL_HOST: "{{ hub_configuration.hostname | mandatory }}"
    application_server_env: "{{ base_server_env | combine (application_server_specific_envs) }}"
  docker_container:
    name: Bla
    volumes: ...
    env: "{{ application_server_env }}"
    # ... more
```